LibVLCSharp/Shared/Helpers/MarshalUtils.cs

```diff
@@ -53,9 +53,118 @@ internal static class MarshalUtils
 
             [DllImport(Constants.Msvcrt, EntryPoint = "vsprintf", CallingConvention = CallingConvention.Cdecl)]
             public static extern int vsprintf_windows(IntPtr buffer, IntPtr format, IntPtr args);
+
+            [DllImport(Constants.libSystem, EntryPoint = "vsnprintf", CallingConvention = CallingConvention.Cdecl)]
+            public static extern int vsnprintf_apple(IntPtr buffer, UIntPtr size, IntPtr format, IntPtr args);
+
+            [DllImport(Constants.Libc, EntryPoint = "vsnprintf", CallingConvention = CallingConvention.Cdecl)]
+            public static extern int vsnprintf_linux(IntPtr buffer, UIntPtr size, IntPtr format, IntPtr args);
+
+            [DllImport(Constants.Msvcrt, EntryPoint = "vsnprintf", CallingConvention = CallingConvention.Cdecl)]
+            public static extern int vsnprintf_windows(IntPtr buffer, UIntPtr size, IntPtr format, IntPtr args);
+        }
+
+        #region logging
+
+        internal static string GetLogMessage(IntPtr format, IntPtr args)
+        {
+            // special marshalling is needed on Linux desktop 64 bits.
+            if (PlatformHelper.IsLinuxDesktop && PlatformHelper.IsX64BitProcess)
+            {
+                return LinuxX64Callback(format, args);
+            }
+
+            var byteLength = vsnprintf(IntPtr.Zero, UIntPtr.Zero, format, args) + 1;
+            if (byteLength <= 1)
+                return string.Empty;
+
+            var buffer = IntPtr.Zero;
+            try
+            {
+                buffer = Marshal.AllocHGlobal(byteLength);
```

Comment:
If the allocation fails, you can't free IntPtr.Zero in the finally block.

Please move the allocation outside the try block.

Reply:
I wasn't aware of that behavior of the FreeHGlobal. OK then...
However, do you think having the Alloc inside the try block helps with the OOME case?

Reply:
It just swallows the exception as this is a non critical failure in a specific scenario, which is the behavior we want IMO.